src/game/scenes/melee.c

```diff
@@ -308,7 +400,58 @@ void handle_action(scene *scene, int player, int action) {
                 }
             }
             break;
+        case ACT_KICK | ACT_UP:
+            if(local->page == PILOT_SELECT && check_pilot_stat_cheat(scene, player) &&
+               local->cheat_pilot_stats_stat[player] > 0) {
+                progressbar_set_highlight(local->bar_stat[player][local->cheat_pilot_stats_stat[player]], 0);
+                local->cheat_pilot_stats_stat[player]--;
+            }
+            break;
+        case ACT_KICK | ACT_DOWN:
+            if(local->page == PILOT_SELECT && check_pilot_stat_cheat(scene, player) &&
+               local->cheat_pilot_stats_stat[player] < 2) {
+                progressbar_set_highlight(local->bar_stat[player][local->cheat_pilot_stats_stat[player]], 0);
+                local->cheat_pilot_stats_stat[player]++;
+            }
+            break;
+        case ACT_KICK | ACT_LEFT:
+            if(local->page == PILOT_SELECT && check_pilot_stat_cheat(scene, player)) {
+                sd_pilot *pilot = game_state_get_player(scene->gs, player)->pilot;
+                int stat = local->cheat_pilot_stats_stat[player];
+                uint8_t *stat_val = get_pilot_stat(pilot, stat);
+                // reallocate points to other stats from selected stat
+                for(int other = 0; other < STAT_COUNT; other++) {
+                    uint8_t *other_val = get_pilot_stat(pilot, other);
+                    if(other == stat || *other_val >= MAX_STAT || *stat_val <= 0)
+                        continue;
+                    (*other_val)++;
+                    (*stat_val)--;
+                }
+                refresh_pilot_stats(scene, player);
+            }
+            break;
+        case ACT_KICK | ACT_RIGHT:
+            if(local->page == PILOT_SELECT && check_pilot_stat_cheat(scene, player)) {
+                sd_pilot *pilot = game_state_get_player(scene->gs, player)->pilot;
+                int stat = local->cheat_pilot_stats_stat[player];
+                uint8_t *stat_val = get_pilot_stat(pilot, stat);
+                // reallocate points from other stats into selected stat
+                for(int other = 0; other < STAT_COUNT; other++) {
+                    uint8_t *other_val = get_pilot_stat(pilot, other);
+                    if(other == stat || *other_val <= 0 || *stat_val >= MAX_STAT)
+                        continue;
+                    (*other_val)--;
+                    (*stat_val)++;
+                }
+                refresh_pilot_stats(scene, player);
+            }
+            break;
         case ACT_KICK:
+            if(local->page == PILOT_SELECT && check_pilot_stat_cheat(scene, player)) {
+                progressbar_set_highlight(local->bar_stat[player][local->cheat_pilot_stats_stat[player]], 1);
+                break;
+            }
+            // [[fallthrough]]
```

Comment:
```suggestion
            // [[fallthrough]]
            __attribute__ ((fallthrough));
```

Reply:
huh, I didn't know that one existed

Reply:
that is a nonstandard extension, not supported by MSVC.

Reply:
Gotta find something that keeps all 3 happy, somehow

Reply:
C23 would be nice, it arguably makes C clearer. Probably not should wait a couple of years to wait for compiler support though.

Reply:
OMF_FALLTHROUGH macro that expands to one of them?
I didn't think it was necessary to explicitly mark it here, but left the comment for us to find when we later support C23.